packages/cashier/src/stores/payment-agent-transfer-store.ts

```diff
@@ -1,54 +1,71 @@
 import { action, computed, observable } from 'mobx';
+import { PaymentAgentDetailsResponse, PaymentAgentTransferResponse } from '@deriv/api-types';
 import { routes } from '@deriv/shared';
 import Constants from 'Constants/constants';
 import ErrorStore from './error-store';
+import {
+    TExtendedPaymentAgentListResponse,
+    TPaymentAgent,
+    TPaymentAgentTransferReceipt,
+    TPaymentAgentTransferConfirm,
+    TPaymentAgentTransferRequest,
+    TRootStore,
+    TTransferLimit,
+    TWebSocket,
+} from 'Types';
 
 export default class PaymentAgentTransferStore {
-    constructor({ WS, root_store }) {
+    constructor(public WS: TWebSocket, public root_store: TRootStore) {
         this.root_store = root_store;
         this.WS = WS;
     }
 
     @observable container = Constants.containers.payment_agent_transfer;
-    @observable error = new ErrorStore();
+    @observable error: TRootStore['modules']['cashier']['error'] = new ErrorStore();
     @observable is_payment_agent = false;
     @observable is_try_transfer_successful = false;
     @observable is_transfer_successful = false;
-    @observable confirm = {};
-    @observable receipt = {};
-    @observable transfer_limit = {};
+    @observable confirm: TPaymentAgentTransferConfirm = {};
+    @observable receipt: TPaymentAgentTransferReceipt = {};
+    @observable transfer_limit: TTransferLimit = {};
+    @observable onRemount: () => void = null;
 
     @computed
-    get is_payment_agent_transfer_visible() {
+    get is_payment_agent_transfer_visible(): boolean {
         return this.is_payment_agent;
     }
 
-    async checkIsPaymentAgent() {
+    async checkIsPaymentAgent(): Promise<void> {
         const get_settings = (await this.WS.authorized.storage.getSettings()).get_settings;
         this.setIsPaymentAgent(get_settings?.is_authenticated_payment_agent ?? false);
     }
 
     @action.bound
-    setIsPaymentAgent(is_payment_agent) {
+    setIsPaymentAgent(is_payment_agent: number | boolean): void {
```

Comment:
Why not only `boolean`?

Reply:
@Bahar-FS because when we call this.setIsPaymentAgent with `get_settings?.is_authenticated_payment_agent`, `is_authenticated_payment_agent` can be 1 or 0,  i refactored the code. 
`this.setIsPaymentAgent(get_settings?.is_authenticated_payment_agent ?? false)` changed to 
`!!get_settings?.is_authenticated_payment_agent`
so now `setIsPaymentAgent` method accepts only boolean values